src/flows/parser_item_html_text_streaming_flow.py

```diff
@@ -41,9 +41,70 @@
 on_error=ABORT_STATEMENT;
 """
 
+DATABASE_NAME = ARTICLES_DB_SCHEMA_DICT[ENVIRONMENT]["articles_database_name"]
+SCHEMA_NAME = ARTICLES_DB_SCHEMA_DICT[ENVIRONMENT]["articles_schema_name"]
+
+# SQL statements to insert new ordered raw data
+# These depend on the initial execution of the 'initial clean' statements in the sql folder
+# This will execute after the copy statement as a single statement
+
+INSERT_ORDERED_SQL = f"""set max_date = (select max(snowflake_loaded_at) from {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_live);
+        
+    insert into {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_live (
+        RESOLVED_ID, 
+                HTML, 
+                TEXT, 
+                SNOWFLAKE_LOADED_AT,
+                TEXT_MD5
+    )
+    select RESOLVED_ID, 
+                HTML, 
+                TEXT, 
+                SNOWFLAKE_LOADED_AT,
+                TEXT_MD5
+    from raw.ITEM.ARTICLE_CONTENT_V2
+        where snowflake_loaded_at > $max_date
+        order by resolved_id;"""
+
+# This will run after an hourly update only once on the weekend
+# Will probably need to apply grants to the renamed live table
+# Revoke grants will remove access to the old live table for better UX
+WEEKLY_DEDUP_SQL = f"""
+    -- weekly dedup on 3XL warehouse
+
+    use warehouse dpt_wh_3xl;
+    
+    create or replace table {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_new as (
+    select RESOLVED_ID, 
+    HTML, 
+    TEXT, 
+    SNOWFLAKE_LOADED_AT,
+    TEXT_MD5,
+    current_timestamp() as last_ordered
+    from {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_live
+    qualify row_number() over (partition by resolved_id order by snowflake_loaded_at desc) = 1
+    order by resolved_id);
+
+    -- clean up ownership
+    grant ownership on table {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_new to role LOADER REVOKE CURRENT GRANTS;
+
+    -- apply proper grants to new ordered table before swap
+
+    grant select, delete on table {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_new to role USER_DATA_DELETION_ROLE;
+    grant all on table {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_new to role ML_SERVICE_ROLE;
+    grant select on table {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_new to role TRANSFORMER;
+    grant select on table {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_new to role SELECT_ALL_ROLE;
+
+    -- swap and drop old table
+    
+    alter table {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_live swap with {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_new;
+    drop table {DATABASE_NAME}.{SCHEMA_NAME}.article_content_ordered_new;
+    
+"""
```

Comment:
The database and schema can be passed to the Prefect task:
```python
PocketSnowflakeQuery()(
    query=WEEKLY_DEDUP_SQL,
    database=config.SNOWFLAKE_ANALYTICS_DATABASE,
    schema=config.SNOWFLAKE_ANALYTICS_DBT_SCHEMA,
)
```
Where possible, I'd like to avoid inserting data into queries using string formatting, and instead use parameterized queries. Another developer who's new to SQL might use the f-string query to insert user-controlled data into a query, and thereby risk SQL-injection.

Reply:
I moved to using the db and schema task args.